csharp/src/Ice/OutgoingRequestFrame.cs

```diff
@@ -84,7 +133,31 @@ public sealed class OutgoingRequestFrame : OutputStream
                     Context[key] = value; // the proxy Context entry prevails.
                 }
             }
-            ContextHelper.Write(this, Context);
+
+            ContextHelper.Write(ostr, Context);
+            _payloadStart = ostr.Tail;
+
+            if (writeEmptyEncaps)
+            {
+                Encoding.CheckSupported();
+                _payloadEnd = ostr.WriteEmptyEncapsulation();
```

Comment:
We need to pass Encoding to WriteEmptyEncapsulation, and arguably it's WriteEmptyEncapsulation that should perform the check.